src/Drivers/Sparse/NlpSparseRajaEx2.cpp

```diff
@@ -0,0 +1,484 @@
+    RAJA_LAMBDA(RAJA::Index_type i)
+    {      
+      // --- constraint 1 body --->  4*x_1 + 2*x_2 == 10
+      cons[0] = 4*x[0] + 2*x[1];
+      // --- constraint 2 body ---> 2*x_1 + x_3
+      cons[1] = 2*x[0] + 1*x[2];
+
+      if(rankdefic_ineq) {
+        // [-inf] <= 4*x_1 + 2*x_3 <= [ 19 ]
+        cons[n-1] = 4*x[0] + 2*x[2];
+      }
+
+      if(rankdefic_eq) {
+        //  4*x_1 + 2*x_2 == 10
+        cons[m-1] = 4*x[0] + 2*x[1];
+      }
+    });
+
+  RAJA::forall<ex2_raja_exec>(RAJA::RangeSegment(2, n-1),
+    RAJA_LAMBDA(RAJA::Index_type i)
+    {
+      // --- constraint 3 body ---> 2*x_1 + 0.5*x_i, for i>=4
+      cons[i] = 2*x[0] + 0.5*x[i+1];
+    });
+
+  return true;
+}
+
+bool SparseRajaEx2::eval_Jac_cons(const size_type& n,
+                                  const size_type& m,
+                                  const double* x,
+                                  bool new_x,
+                                  const int& nnzJacS,
+                                  index_type* iJacS,
+                                  index_type* jJacS,
+                                  double* MJacS)
+{
+  assert(n==n_vars_); assert(m==n_cons_);
+  assert(n>=3);
+
+  assert(nnzJacS == 4 + 2*(n-3) + 2*rankdefic_eq_ + 2*rankdefic_ineq_);
+
+  // RAJA doesn't like member objects
+  bool rankdefic_eq = rankdefic_eq_;
+  bool rankdefic_ineq = rankdefic_ineq_;
+
+  if(iJacS !=nullptr && jJacS != nullptr) {
+    // serial part
+    RAJA::forall<ex2_raja_exec>(RAJA::RangeSegment(0, 1),
+      RAJA_LAMBDA(RAJA::Index_type itrow)
+      {
+        // --- constraint 1 body --->  4*x_1 + 2*x_2 == 10
+        iJacS[0] = 0;
+        jJacS[0] = 0;
+        iJacS[1] = 0;
+        jJacS[1] = 1;
+        // --- constraint 2 body ---> 2*x_1 + x_3
+        iJacS[2] = 1;
+        jJacS[2] = 0;
+        iJacS[3] = 1;
+        jJacS[3] = 2;
+
+        if(rankdefic_ineq) {
+          // [-inf] <= 4*x_1 + 2*x_3 <= [ 19 ]
+          iJacS[2*n-2] = n-1;
+          jJacS[2*n-2] = 0;
+          iJacS[2*n-1] = n-1;
+          jJacS[2*n-1] = 2;
+        }
+
+        if(rankdefic_eq) {
+          //  4*x_1 + 2*x_2 == 10
+          iJacS[2*m-2] = m-1;
+          jJacS[2*m-2] = 0;
+          iJacS[2*m-1] = m-1;
+          jJacS[2*m-1] = 1;
+        }
+      });
+    
+    RAJA::forall<ex2_raja_exec>(RAJA::RangeSegment(2, n-1),
+      RAJA_LAMBDA(RAJA::Index_type itrow)
+      {
+        // --- constraint 3 body ---> 2*x_1 + 0.5*x_i, for i>=4
+        iJacS[2*itrow] = itrow;
+        jJacS[2*itrow] = 0;
+        iJacS[2*itrow+1] = itrow;
+        jJacS[2*itrow+1] = itrow+1;
+      });
+
+  }
+
+  if(MJacS != nullptr) {
+    // serial part
+    RAJA::forall<ex2_raja_exec>(RAJA::RangeSegment(0, 1),
+      RAJA_LAMBDA(RAJA::Index_type itrow)
+      {
+        // --- constraint 1 body --->  4*x_1 + 2*x_2 == 10
+        MJacS[0] = 4.0;
+        MJacS[1] = 2.0;
+        // --- constraint 2 body ---> 2*x_1 + x_3
+        MJacS[2] = 2.0;
+        MJacS[3] = 1.0;
```

Comment:
I am not sure this makes sense. Consider rewriting as
 ```suggestion
        if(itrow == 0) {
          MJacS[0] = 4.0;
          MJacS[1] = 2.0;
          // --- constraint 2 body ---> 2*x_1 + x_3
          MJacS[2] = 2.0;
          MJacS[3] = 1.0;
        }
```

Reply:
Actually, this is better added on the host and then copied to GPU, since these are just constant Jacobian elements that stay the same throughout the computation.

Reply:
`if` condition is not required here, since here `RAJA::RangeSegment` only has one valid number.